In [2]:
# Descarga y preparación de datos
# Importación de librerías
import pandas as pd

In [3]:
# Carga de datos y visualización de las primeras filas
Churn_data = pd.read_csv('Churn.csv')
Churn_data.head()

,RowNumber,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,1,15634602,Hargrave,619,France,Female,42,2.0,0.00,1,1,1,101348.88,1
1,2,15647311,Hill,608,Spain,Female,41,1.0,83807.86,1,0,1,112542.58,0
2,3,15619304,Onio,502,France,Female,42,8.0,159660.80,3,1,0,113931.57,1
3,4,15701354,Boni,699,France,Female,39,1.0,0.00,2,0,0,93826.63,0
4,5,15737888,Mitchell,850,Spain,Female,43,2.0,125510.82,1,1,1,79084.10,0


In [4]:
# Información general del DataFrame
Churn_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   RowNumber        10000 non-null  int64  
 1   CustomerId       10000 non-null  int64  
 2   Surname          10000 non-null  object 
 3   CreditScore      10000 non-null  int64  
 4   Geography        10000 non-null  object 
 5   Gender           10000 non-null  object 
 6   Age              10000 non-null  int64  
 7   Tenure           9091 non-null   float64
 8   Balance          10000 non-null  float64
 9   NumOfProducts    10000 non-null  int64  
 10  HasCrCard        10000 non-null  int64  
 11  IsActiveMember   10000 non-null  int64  
 12  EstimatedSalary  10000 non-null  float64
 13  Exited           10000 non-null  int64  
dtypes: float64(3), int64(8), object(3)
memory usage: 1.1+ MB


In [5]:
# Número de filas y columnas
print(f'Número de filas: {Churn_data.shape[0]}, Número de columnas: {Churn_data.shape[1]}')

Número de filas: 10000, Número de columnas: 14


In [6]:
# Descripción del DataFrame
print(Churn_data.describe())

         RowNumber    CustomerId   CreditScore           Age       Tenure  \
count  10000.00000  1.000000e+04  10000.000000  10000.000000  9091.000000   
mean    5000.50000  1.569094e+07    650.528800     38.921800     4.997690   
std     2886.89568  7.193619e+04     96.653299     10.487806     2.894723   
min        1.00000  1.556570e+07    350.000000     18.000000     0.000000   
25%     2500.75000  1.562853e+07    584.000000     32.000000     2.000000   
50%     5000.50000  1.569074e+07    652.000000     37.000000     5.000000   
75%     7500.25000  1.575323e+07    718.000000     44.000000     7.000000   
max    10000.00000  1.581569e+07    850.000000     92.000000    10.000000   

             Balance  NumOfProducts    HasCrCard  IsActiveMember  \
count   10000.000000   10000.000000  10000.00000    10000.000000   
mean    76485.889288       1.530200      0.70550        0.515100   
std     62397.405202       0.581654      0.45584        0.499797   
min         0.000000       1.00000

In [7]:
# Buscamos y contamos valores nulos
Churn_data.isna().sum()

RowNumber            0
CustomerId           0
Surname              0
CreditScore          0
Geography            0
Gender               0
Age                  0
Tenure             909
Balance              0
NumOfProducts        0
HasCrCard            0
IsActiveMember       0
EstimatedSalary      0
Exited               0
dtype: int64

En la columna 'Tenure' tenemos 909 filas sin algún valor, veremos más adelante que haremos con ellos, porque representan el periodo en que estuvieron los clientes con el servicio

In [8]:
# Buscaremos valores duplicados
print(f'Número de filas duplicadas: {Churn_data.duplicated().sum()}')

Número de filas duplicadas: 0


In [9]:
# Clasificaremos las columnas por tipo de dato para separarlas en numéricas y categóricas
# Separamos las columnas que no aportan valor para nuestro análisis
no_informativas_cols = ['RowNumber', 'CustomerId', 'Surname']

# Columnas Numéricas
numericas_cols = ['CreditScore', 'Age', 'Tenure', 'Balance', 'NumOfProducts', 'HasCrCard', 'IsActiveMember', 'EstimatedSalary']

# Columnas Categóricas
categoricas_cols = ['Geography', 'Gender', 'Exited']

In [10]:
# Eliminaremos las columnas no informativas del DataFrame
Churn_data = Churn_data.drop(no_informativas_cols, axis=1)

In [11]:
# Transformaremos las variables categóricas en numéricas utilizando One-Hot Encoding
Churn_data = pd.get_dummies(Churn_data, drop_first=True, columns=categoricas_cols)

In [12]:
Churn_data.head()

,CreditScore,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Geography_Germany,Geography_Spain,Gender_Male,Exited_1
0,619,42,2.0,0.00,1,1,1,101348.88,False,False,False,True
1,608,41,1.0,83807.86,1,0,1,112542.58,False,True,False,False
2,502,42,8.0,159660.80,3,1,0,113931.57,False,False,False,True
3,699,39,1.0,0.00,2,0,0,93826.63,False,False,False,False
4,850,43,2.0,125510.82,1,1,1,79084.10,False,True,False,False


In [ ]:
# Estamos observando el desbalance de esta columna para saber si es necesario aplicar técnicas de balanceo de datos
Churn_data['Exited_1'].value_counts()

Exited_1
False    7963
True     2037
Name: count, dtype: int64